docs/creating-nodes/testing.md

```diff
@@ -0,0 +1,271 @@
+```
+
+Install modules like `sinon` and `nock` as needed.
+
+### 2. Creating the test code
+
+Create a spec file that contains the test directory and the test code.
+You would typically create a `test` directory in the same directory as package.json, and use the naming rule `file-to-be-tested_spec.js` for the test files themselves.
+[Example]
+
+```text
+.
+├── test-node.js
+├── test-node.html
+├── package.json
+└── test
+    └── test-node_spec.js  // As the spec file for test-node.js, the file name is test-node_spec.js
+```
+
+The processing of test code typically follows this sequence:
+
+1. Load libraries
+2. Initialize helper
+3. Start helper
+4. Load test flow into helper
+5. Run test flow in helper
+6. Evaluate processing results
+7. Exit helper
+
+The following explains how to create test code based on the [example](https://github.com/node-red/node-red-node-test-helper#Example-unit-test) in the helper.
+The test code is written using Mocha.
+Specific details about Mocha are not given here, but you can learn about writing test code in Mocha by visiting the preceding URL and other resources.
+
+```javascript
+// 1. Load libraries
+var should = require("should");
+var helper = require("node-red-node-test-helper");
+var lowerNode = require("../lower-case.js");
+
+// 2. Initialize helper
+helper.init(require.resolve('node-red'));
+
+describe('myNode', function () {
+
+  // 3. Start helper
+  beforeEach(function (done) {
+      helper.startServer(done);
+  });
+
+  // 7. Exit helper
+  afterEach(function (done) {
+      helper.unload();
+      helper.stopServer(done);
+  });
+
+  // Test case
+  it('should make payload lower case', function (done) {
+    // Define test flow
+    var flow = [{ id: "f1", type:"tab"},
+                { id: "n1", z:"f1", type: "lower-case", wires:[["n2"]] },
+                { id: "n2", z:"f1", type: "helper" }];
+    // 4. Load test flow into helper
+    helper.load(lowerNode, flow, function () {
+      var lowerCase = helper.getNode("n1");
+      var helperNode = helper.getNode("n2");
+      // 6. Evaluate processing results
+      helperNode.on("input", function (msg) {
+        msg.should.have.property('payload', 'uppercase');
+        done();
+      });
+      // 5. Run test flow in helper
+      lowerCase.receive({ payload: "UpperCase" });
+    });
+  });
+
+  it('should do something', function (done) {
+    // Write test codes here ...
+  });
+});
+```
+
+#### 1. Load libraries
+
+As a minimum, you will need to load the following libraries:
+
+1. should.js
+2. node test helper
+3. The node to be tested (lower-case.js in the example)
+
+Add libraries like sinon.js as needed.
+
+#### 2. Initialize helper
+
+The helper does not in itself have the ability to run nodes and does so by referencing code in the Node-RED platform.
+Give helper access to the code of the Node-RED platform by passing the Node-RED path to `helper.init()`.
+
+#### 3. Start helper and 7. Exit helper
```

Comment:
Why is 7 here ? - keep it in sequence even if repetitive.

Reply:
I fixed it in the commit, https://github.com/node-red/node-red.github.io/pull/109/commits/21c53366ea9ddb3cce5006b7403bfb3674df75b6 .